meson.build

```diff
@@ -34,6 +34,9 @@ if cc.get_id() == 'clang'
 	add_project_arguments('-Wno-missing-braces', language: 'c')
 endif
 
+# Wayland deprecated wl_buffer, Mesa EGL still uses it, at least on FreeBSD
+add_project_arguments('-Wno-error=deprecated-declarations', language: 'c')
```

Comment:
Can you only add this on FreeBSD?

Reply:
Are you sure you can't see that on Linux? Does `EGL/eglmesaext.h` not mention `wl_buffer`?

Reply:
It does, but it also doesn't raise a deprecation error. We would have heard about it by now.

Reply:
Weird… hmm, I guess your compiler does not include `deprecated-declarations` in the default warning set! clang 5.0.0 has a lot more warnings enabled by default

Reply:
We run builds on Clang, too, so... https://builds.sr.ht/job/320

Reply:
I think that only the wl_buffer inside wayland-server.h (where you can see its actual declaration) is deprecated.
The opaque wl_buffer from wayland-server/client-protocol.h works fine.
Add -DWL_HIDE_DEPRECATED to the build system or something.

Reply:
Yeah, that works, thanks @ascent12